src/Microsoft.OData.Core/Json/BufferingJsonReader.cs

```diff
@@ -81,6 +81,64 @@ internal BufferingJsonReader(IJsonReader innerReader, string inStreamErrorProper
             this.currentBufferedNode = null;
         }
 
+        internal bool TryGetValueFromBuffered(string propertyName, bool removeIfExist, out object value)
+        {
+            bool found = false;
+            BufferedNode currentNode = this.bufferedNodesHead;
+            if (this.bufferedNodesHead != null)
+            {
+                do
+                {
+                    if (currentNode.NodeType == JsonNodeType.StartObject ||
+                        currentNode.NodeType == JsonNodeType.StartArray)
+                    {
+                        break;
+                    }
+
+                    if (currentNode.NodeType == JsonNodeType.Property &&
+                        currentNode.Value.ToString() == propertyName &&
+                        currentNode.Next != currentNode) // must have another node
+                    {
+                        found = true;
+                        break;
+                    }
+
+                    currentNode = currentNode.Next;
+                }
+                while (currentNode != this.bufferedNodesHead);

```

Comment:
Was wondering how `currentNode` ever be `bufferedNodeHead` then I confirmed that this is a circular linked list. I think we should abstract traversal and access of the nodes through some helper method or inner class to avoid accidentally accessing the nodes the wrong way.

Reply:
I checked the logics in 'BefferingJsonReader' and that's the way to check the end of link.